### ISSUE TITLE
Docs don't compile

### ISSUE THREAD

**Issue by jcjaskula** [enthought/traits#208]
I'm unable to build the docs with sphinx either from setup.py or directly from sphinx. Apparently robots are also getting errors since few months:

https://readthedocs.org/builds/traits/1453057/

Is it because of a missing dependency, a new version of sphinx or outdated docs? I'm trying to build them to include them in debian packages.

Cheers,

```
~/traits-4.5.0$ python setup.py build
... Ok
$ python setup.py build_sphinx
running build_sphinx
Running Sphinx v1.2.3
Traceback (most recent call last):
  File "setup.py", line 64, in <module>
    cmdclass=additional_commands(),
  File "/usr/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/usr/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/usr/lib/python2.7/dist-packages/sphinx/setup_command.py", line 158, in run
    freshenv=self.fresh_env)
  File "/usr/lib/python2.7/dist-packages/sphinx/application.py", line 119, in __init__
    self.setup_extension(extension)
  File "/usr/lib/python2.7/dist-packages/sphinx/application.py", line 336, in setup_extension
    err)
sphinx.errors.ExtensionError: Could not import extension traits.util.trait_documenter (exception: cannot import name ctraits)
```

```
~/traits-4.5.0/docs$ make html
sphinx-build -b html -d build/doctrees   source build/html
Running Sphinx v1.2.3
loading pickled environment... not yet created
building [html]: targets for 31 source files that are out of date
updating environment: 31 added, 0 changed, 0 removed
reading sources... [ 12%] traits_api_reference/has_dynamic_views                                                                                              
Exception occurred:
  File "/usr/lib/python2.7/tokenize.py", line 190, in add_whitespace
    .format(row, col, self.prev_row, self.prev_col))
ValueError: start (0,15) precedes previous end (1,0)
```

**Comment by mdickinson** [comment]
I apologise for taking so long to get to this.  What's happening is that we have a long-standing bug in our own Sphinx plugin, namely that during parsing of Traits-specific documentation, we create a list of tokens with line number 0 (which should be line number 1 instead).  That bug wasn't exposed until Python's tokenize module received a "fix" that make error-checking more strict (see http://bugs.python.org/issue12691).  That fix first appeared in Python 2.7.7.

I'll open a PR to fix the bug in our source.

The broken Read the Docs build looks like a separate issue, to do with RtD not being able to build C extensions and the Sphinx autodoc build requiring them.

**Comment by mdickinson** [comment]
Fix in enthought/traits#232.

**Comment by jonathanrocher** [comment]
Ready to close.
